[Alpha] Adding support for validating venues

davidakachaos opened the pull request:
Adding basic support for checking Venues like we currently check segments.
The only active check is a check if a venue has a city set.

Note: I just blindingly copied the functions of segments to venues. Will need some work of course, but for now this works. I did already reworked some, but maybe there are things that can be way more efficient than they are now...

berestovskyy commented:
In general, looks fine. I propose to merge it into a "places" or a "dev" branch, not master and continue the development there for a while.

One things IMO need to be addressed: we need to avoid function duplication. It will be hard to support etc etc. See [DRY principle](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) for more details.

What if we adapt "simple segment" concept to a "simple object"? Both segments and places have a lot of attributes in common right now. Plus we might rename some, like `$segmentID` --> `$objID`, plus we might add some like `$objType`...

What do you think?

davidakachaos commented:
I totally agree with keeping it DRY and the adaptions to SimpleSEGMENT to SimpleOBJECT so we can remove a lot of redundant code.
Merging this to a "places" or "dev" branch seems like a good thing to me. Let's make that happen and continue development in that branch for now.

berestovskyy commented:
No, just rebase the changes to that branch (if necessary) and edit this request, so we just merge it as is to a new branch.

davidakachaos commented:
@berestovskyy I've removed the duplicate code for the venues, changed SimpleSEGMENT to SimpleOBJECT to accommodate the venues in it and tested these locally and they seem to work.
Shall we merge this PR to the dev branch to make it up to date? This seems a good enough base for development.

davidakachaos commented on src/validate.js:
```diff
@@ -428,8 +433,21 @@ function F_VALIDATE(disabledHL) {
+		/** @type {string} */
+		this.$brand = "";
+		/** @type {string} */
+		this.$objectID = objID;
```
The ID of a venue within WME is a dot seperated string containing 3 numbers. Like `"182518064.1825246174.289659"`

Why that is? I don't know, but that are the IDs.

berestovskyy replied:
Sorry, I did not know that. So, if the venue ID is a string, but the segmentID is a number, we should use pluses (`+`) very carefully, because `+"1.2.3"` is always `NaN`.

The type of objectID should be `string|number` I guess. With this new info, I will thoroughly look through the changes again...

davidakachaos replied:
Yes those sneaky little pluses before the variable name..... And I was wondering why my strings "suddenly" broke! We both learned haha. I'm looking forward to your comments, I did rework things a bit already.

davidakachaos commented on src/report.js:
```diff
@@ -1223,10 +1127,10 @@ function F_SHOWREPORT(reportFormat) {
-		if(obj.$segmentCopy){
-			if (50 > obj.$segmentCopy.$length)
+		if(obj.$objectCopy){
+			if (50 > obj.$objectCopy.$length)
```
One i guess? Or is just count of venues on a given street?

berestovskyy commented:
It is getting a bit messy. Please make the changes you think are appropriate, squash and merge this pull request to the `dev` branch. We will add more on top of that.

Thanks for such a great job!

davidakachaos commented:
I'll squash and merge this as is and continue the work on it.

Thanks for your support! Learning to code better every step 👍
